### DISCUSSION TITLE
Listing/viewing collections does not return a proper status code for unauthenticated users

### DISCUSSION THREAD

**Discussion by antonioggeorgiev** [pocketbase/pocketbase#1046] [Q&A]
I am using React with react-router-dom. I have created some protected routes which can be accessed only if there is an authenticated user. I have created an interceptor, that listens for status 403 and invalidates the user (in the state). From the documentation, I saw that this rule - @request.user.id != "" is for 'registered users'.  However, even when I am not authenticated, I receive a list of records but empty. I want somehow to receive a status code of 403 when I try something that I am not authorized for. 
Is there a proper way of doing that? Otherwise, even if the token has expired, I won't be able to properly update the state and force the user to re-authenticate.

**Comment by ganigeorgiev** [comment]
PocketBase API rules act both as access and filters control and there is no easy way (without extra parsing and static analysis) to know upfront what actually your rule do and what exact part/expression will result to `false`. And therefore for the list/search action we return an empty items list.

> I want somehow to receive a status code of 403 when I try something that I am not authorized for.

You can do that only if you use [PocketBase as framework](https://pocketbase.io/docs/use-as-framework/) and attach for example one of the exposed internal middlewares like `apis.RequireRecordAuth()` (_more documentation how to extend PocketBase will be available with the final v0.8.0 release later this week_).

> Is there a proper way of doing that? Otherwise, even if the token has expired, I won't be able to properly update the state and force the user to re-authenticate.

Usually if all rules are properly set you wouldn't end up in a situations where the user will have direct access from the UI to resources they are not authorized. To minimize the rare cases where the app user has access to a resource but their auth token has expired, you can try to call `pb.collection("users").authRefresh()` on each page reload/app start to get a new token. This call will return a 401 if the existing auth token is expired/invalid/missing (and 403 if the current auth record is not from the "users" collection).
